apps/hpc-ftsadmin/src/app/components/flow-form/flow-form.tsx

```diff
@@ -0,0 +1,2295 @@
+        },
+      })
+      .then(() => {
+        toast.success(
+          t.t(
+            lang,
+            (s) => s.components.flowForm.submitValidation.rejectSuccess
+          ),
+          TOAST_CONFIG
+        );
+        load();
+      })
+      .catch((error) => {
+        toast.error(
+          isCustomError(error)
+            ? error.json.message
+            : t.t(
+                lang,
+                (s) => s.components.flowForm.submitValidation.unknownError
+              ),
+          TOAST_CONFIG_ERROR
+        );
+      })
+      .finally(() => setShouldRejectLoading(false));
+  };
+  const handleApproveAll = () => {
+    setShouldAcceptAllPendingChanges(true);
+  };
+
+  const handleRejectAll = () => {
+    setPendingValues(null);
+    setPendingValuesHandled(0);
+  };
+
+  const handleFundingDestinationOrganizations = (
+    newValue: util.FormObjectValue | util.FormObjectValue[] | null,
+    setFieldValue: FormikHelpers<FlowFormType>['setFieldValue']
+  ) => {
+    setFieldValue('fundingDestinationOrganizations', newValue);
+    if (!Array.isArray(newValue)) {
+      return;
+    }
+
+    if (!newValue.some((org) => org.isConfidential)) {
+      setFieldValue('fundingDestinationAnonymizedOrganizations', []);
+    }
+  };
+
+  const handleMethod = (
+    newValue: util.FormObjectValue | util.FormObjectValue[] | null,
+    setFieldValue: FormikHelpers<FlowFormType>['setFieldValue']
+  ) => {
+    setFieldValue('method', newValue);
+
+    //  ChildMethod is `FormObjectValue`
+    if (Array.isArray(newValue) || typeof newValue === 'string') {
+      return;
+    }
+
+    setFieldValue('childMethod', null);
+  };
+  return (
+    <Formik
+      initialValues={flowInitialValues}
+      onSubmit={(values) => handleSubmit(values as FlowFormTypeValidated)}
+      validate={(values) =>
+        validateForm(
+          values,
+          FORM_VALIDATION_SCHEMA,
+          validationErrorMessages(lang)
+        )
+      }
+    >
+      {({ values, isValid: formikValid, setFieldValue, dirty }) => {
+        const isValid =
+          formikValid &&
+          values.reportingDetails.reduce(
+            (acc, { reportedByOrganization, reportChannel, url }) =>
+              acc &&
+              !validateReportingDetailsRequiredField(
+                reportedByOrganization,
+                lang
+              ) &&
+              !validateReportingDetailsRequiredField(reportChannel, lang) &&
+              !validateReportingDetailsURLFormat(url, lang),
+            true
+          );
+        const isFlowLinkEmptyAndDisabled =
+          values.childFlows.length === 0 && !values.parentFlow && isDisabled;
+        return (
+          <Form>
+            <BlockNavigationOnUnsavedChanges
+              {...{
+                dirty,
+                submitLoading: isSubmitLoading,
+                rejectLoading: shouldRejectLoading,
+                deleteLoading: isDeleteLoading,
+              }}
+            />
```

Comment:
Any specific reason why this is not
```suggestion
            <BlockNavigationOnUnsavedChanges
              dirty={dirty}
              submitLoading={isSubmitLoading}
              rejectLoading={shouldRejectLoading}
              deleteLoading={isDeleteLoading}
            />
```

Reply:
Prior to renaming boolean variables to should... is... has..., etc. It should have looked like 
```ts
 {...{
      dirty,
      submitLoading,
      rejectLoading,
      deleteLoading,
   }}
```

I hate to have to write  `dirty={dirty}` when it has the same name, so I wanted to try this alternative. So I might rename `BlockNavigationOnUnsavedChanges` props instead.